test/plots/stackedBarPlotTests.ts

```diff
@@ -1,607 +1,411 @@
+      it("renders rects offset by previous values", () => {
+        let bars = stackedBarPlot.content().selectAll("rect");
+
+        let dataLength = stackedBarPlot.datasets()[0].data().length;
+        let dataCount = stackedBarPlot.datasets().length * dataLength;
+        assert.strictEqual(bars.size(), dataCount, "same number of bars as data");
+
+        let calculateStackedYs = (yAccessor: Plottable.Accessor<number>) => {
+          let stackedYDataArray: number[][] = [];
+          stackedBarPlot.datasets().forEach((dataset, datasetIndex) => {
+            let yData = dataset.data().map((d, i) => yAccessor(d, i, dataset));
+            if (datasetIndex === 0) {
+              stackedYDataArray[datasetIndex] = yData;
+              return;
+            }
+            stackedYDataArray[datasetIndex] = dataset.data().map((d, i) => {
+              return yData[i] + stackedYDataArray[datasetIndex - 1][i];
+            });
+          });
+          return stackedYDataArray;
+        };
+
+        let stackedYs = calculateStackedYs((d) => d.y);
+        bars.each(function(d, i) {
+          let bar = d3.select(this);
+          let datasetIndex = Math.floor(i / dataLength);
+          let datumIndex = i % dataLength;
+          assert.closeTo(TestMethods.numAttr(bar, "y"), yScale.scale(stackedYs[datasetIndex][datumIndex]),
+            window.Pixel_CloseTo_Requirement, "y attribute offset set correctly");
+        });
+
+        stackedBarPlot.destroy();
+        svg.remove();
       });
 
-      let dataset2Rects = plot.content().selectAll("." + dataset2.metadata().id);
-      dataset2Rects.each(function (datum, index) {
-        let rect = d3.select(this);
-        let expectedY = yScale.scale(data2[index].y + data1[index].y);
-        let actualY = TestMethods.numAttr(rect, "y");
-        assert.closeTo(actualY, expectedY, 0.1, "y attribute set correctly (dataset 2, datum " + index + ")");
-        let expectedHeight = yScale.scale(data1[index].y) - expectedY;
-        let actualHeight = TestMethods.numAttr(rect, "height");
-        assert.closeTo(actualHeight, expectedHeight, 0.1, "height attribute set correctly (dataset 2, datum " + index + ")");
+      // HACKHACK #2795: correct off-bar label logic to be implemented
+      it("doesn't show any off-bar labels", () => {
+        stackedBarPlot.labelsEnabled(true);
+        yScale.domain([0, 30]);
+        let offBarLabels = stackedBarPlot.content().selectAll(".off-bar-label");
+        assert.operator(offBarLabels.size(), ">", 0, "some off-bar labels are drawn");
+        let offBarLabelVisibilities: string[] = [];
+        offBarLabels.each(function() {
+          offBarLabelVisibilities.push(d3.select(this).style("visibility"));
+        });
+        assert.isTrue(offBarLabelVisibilities.every((visibility) => visibility === "hidden"), "all off-bar labels are hidden");
```

Comment:
is there any reason why you didn't test `visibility === "hidden"` in `each` above but push them to an array then test `every`?